syllabus/examples/experimental/sb3_procgen_plr.py

```diff
@@ -45,21 +217,26 @@ class CustomCallback(BaseCallback):
     def __init__(self, curriculum, verbose=0):
         super().__init__(verbose)
         self.curriculum = curriculum
+        self.global_step = 0
 
     def _on_step(self) -> bool:
-        tasks = self.training_env.venv.venv.venv.get_attr("task")
-
-        update = {
-            "update_type": "on_demand",
-            "metrics": {
-                "value": self.locals["values"],
-                "next_value": self.locals["values"],
-                "rew": self.locals["rewards"],
-                "dones": self.locals["dones"],
-                "tasks": tasks,
-            },
-        }
-        self.curriculum.update_curriculum(update)
+        if args.curriculum and args.curriculum_method == "plr":
+            tasks = self.training_env.venv.venv.venv.get_attr("task")
+
+            update = {
+                "update_type": "on_demand",
+                "metrics": {
+                    "value": self.locals["values"],
+                    "next_value": self.locals["values"],
+                    "rew": self.locals["rewards"],
+                    "dones": self.locals["dones"],
+                    "tasks": tasks,
+                },
+            }
+            self.curriculum.update_curriculum(update)
+        self.global_step += args.num_steps * args.num_envs
+        mean_eval_returns, _, _ = level_replay_evaluate_sb3(args.env_id, model, args.num_eval_episodes, num_levels=0)
+        writer.add_scalar("test_eval/mean_episode_return", mean_eval_returns, self.global_step)
```

Comment:
This code should only be run once every update. There is a different callback method `_on_training_end` that you should probably use.

If you need access to any data from training, try printing out self.locals or self.globals from within the callback method to see what is available